app/src/main/java/com/concordia/mcga/fragments/TransportButtonFragment.java

```diff
@@ -133,22 +158,35 @@ public void expandFAB() {
             publicTransportTextView.startAnimation(transport_textview_open);
             publicTransportFAB.setClickable(true);
 
+
             shuttleFAB.startAnimation(transport_fab_open);
-            shuttleTextView.startAnimation(transport_textview_open);
-            shuttleFAB.setClickable(true);
+            if (shuttleVisible) {
+                shuttleTextView.startAnimation(transport_textview_open);
+                shuttleFAB.setClickable(true);
+            }
+
+            swapIcons(R.drawable.ic_close_black_24dp);
         }
     }
 
     private void swapIcons(int iconId) {
-        FloatingActionButton transportExpandFAB = (FloatingActionButton) getView().findViewById(R.id.transportExpandFAB);
         transportExpandFAB.setImageResource(iconId);
+
+        if (iconId != R.drawable.ic_close_black_24dp)
+            transportExpandFAB.setTag(iconId);
+
+        transportExpandFAB.clearColorFilter();
+    }
+
+    private void restorePreviousIcon() {
+        transportExpandFAB.setImageResource((Integer) transportExpandFAB.getTag());
         transportExpandFAB.clearColorFilter();
     }
 
     private String formatTime(int hours, int minutes) {
```

Comment:
@arekmano 
I meant in the output,
 To print a string like 2h39m instead of 2 hours 39 minutes.

I find it too verbose, and we don't have lo much space on the screen.